tests/sles4sap/hana_test.pm

```diff
@@ -32,12 +36,13 @@ sub run {
     my $sid         = get_required_var('INSTANCE_SID');
     my $instance_id = get_required_var('INSTANCE_ID');
     my $sapadm      = $self->set_sap_info($sid, $instance_id);
+
+    # Test PIDs max, as SAP as some prerequisites on this and change for SAP user
     $self->test_pids_max;
     $self->user_change;
 
     # Check HDB with a database query
-    my $password = get_required_var('PASSWORD');
-    my $output   = script_output "hdbsql -j -d $sid -u SYSTEM -n localhost:30015 -p $password 'SELECT * FROM DUMMY'";
+    my $output = script_output "hdbsql -j -d $sid -u SYSTEM -i $instance_id -p $sles4sap::instance_password 'SELECT * FROM DUMMY'";
```

Comment:
Isn't it better to leave `-n localhost:3${instance_id}15` as part of the command line or is the same as `-i $instance_id` will always attempt to connect to localhost in the port `3${instance_id}15`?

Reply:
Yes it's the same. I found this yesterday and tested on my lab. Adding `-i $instance_id` automatically creates port number with the good value. And if no server is set then `localhost` is used by default.